packages/ckeditor5-ui/src/toolbar/balloon/balloontoolbar.js

```diff
@@ -356,6 +364,23 @@ function getBalloonPositions( isBackward ) {
 	];
 }
 
+function selectionContainsOnlyMultipleObjects( selection, schema ) {
+	// It doesn't contain multiple objects if there is only one range.
+	if ( selection.rangeCount <= 1 ) {
```

Comment:
A `DocumentSelection` in the model will always have at least one range. AFAIR there's no way to force an editor into a range-less selection state. Only virtual `Selection` instances can be range-less.